### PR TITLE
Discrete product rule unit test

### PR DESCRIPTION
**Pull request by akshaysridhar**
Discrete product rule unit tests. 

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

File: test/Operators/hybrid/2d.jl [hunk 1]
```diff
@@ -150,6 +150,49 @@ end
...
+        hv_center_space, hv_face_space = hvspace_2D(helem = n, velem = n)
+        ᶠz = Fields.coordinate_field(hv_face_space).z
+        ᶜz = Fields.coordinate_field(hv_center_space).z
+        Δh[k] = 1.0 / n
+
+        # advective velocity
+        a = Geometry.WVector.(ones(Float64, hv_face_space) .* ᶠz,)
+        # scalar-valued field to be advected
+        b = sin.(ᶠz)
+        c = sin.(ᶜz)
```
Comment (valeriabarra) [inline test/Operators/hybrid/2d.jl:180]: why do you need `c` here? Everything else looks good to me. Thanks for adding these!

**Comment by bischtob** [comment]
Is there a reason that this has not been merged?

**Comment by bischtob** [comment]
@simonbyrne

**Review by valeriabarra — APPROVED** [review]

File: examples/hybrid/plane/test_isothermal_channel.jl [hunk 2]
```diff
@@ -0,0 +1,353 @@
+
```
Comment (simonbyrne) [inline examples/hybrid/plane/test_isothermal_channel.jl:1]: this file isn't being run as part of the tests.

File: examples/hybrid/plane/test_isothermal_channel.jl [hunk 3]
```diff
@@ -0,0 +1,353 @@
...
+push!(LOAD_PATH, joinpath(@__DIR__, "..", ".."))
```
Comment (simonbyrne) [inline examples/hybrid/plane/test_isothermal_channel.jl:2]:
```suggestion
```

File: test/Operators/hybrid/2d.jl [hunk 4]
```diff
@@ -150,6 +150,49 @@ end
...
+    err, Δh = zeros(length(n_elems_seq)), zeros(length(n_elems_seq))
+
+    for (k, n) in enumerate(n_elems_seq)
+        # Discrete Prodouct Rule Test
+        # ∂(ab)/∂s = a̅∂b/∂s + b̅∂a∂s
+        # a, b are interface variables, and  ̅ represents interpolation
+        # c is a cell centered variable
+        # s is the coordinate along horizontal surfaces (terrain following)
+        # For this test, we use a(z) = z and b = sin(z),
+        hv_center_space, hv_face_space = hvspace_2D(helem = n, velem = n)
```
Comment (simonbyrne) [inline test/Operators/hybrid/2d.jl:171]: can we test with a stretched mesh? a lot of issues only show up when the vertical isn't uniforrm

**Comment by simonbyrne** [comment]
@akshaysridhar can you update this?

File: examples/hybrid/plane/test_isothermal_channel.jl [hunk 5]
```diff
@@ -0,0 +1,353 @@
...
+function warp_surface(coord)
+  # Parameters from GMD-9-2007-2016
+  # Specification for Agnesi Mountain following 
+  # Ulrich and Guerra [2016 GMD]
+  x = Geometry.component(coord,1)
+  FT = eltype(x)
+  ac = 5000
+  hc = 500.0
+  h = hc / (1 + ((x-12500)/ac)^2) #+ hc / (1 + ((x - 14000.0)/ac)^2)
+  return h * FT(0)
```
[the remark this answers is not in the document]
Comment (akshaysridhar) [reply examples/hybrid/plane/test_isothermal_channel.jl]: Applied formatter

File: examples/hybrid/plane/test_isothermal_channel.jl (hunk 3 above)
Comment (akshaysridhar) [reply examples/hybrid/plane/test_isothermal_channel.jl]: Removed.

File: test/Operators/hybrid/2d.jl (hunk 1 above)
Comment (akshaysridhar) [reply test/Operators/hybrid/2d.jl:180]: Removed, c was originally included as a variable computed on cell-centers (it remains unused); removed now such that `a,b` are face variables.

File: examples/hybrid/plane/test_isothermal_channel.jl (hunk 2 above)
Comment (akshaysridhar) [reply examples/hybrid/plane/test_isothermal_channel.jl]: Removed. Will push as separate pr

File: test/Operators/hybrid/2d.jl (hunk 4 above)
Comment (akshaysridhar) [reply test/Operators/hybrid/2d.jl:171]: added stretched test

**Comment by akshaysridhar** [comment]
bors try

**Pull request merged**